src/theory/sets/theory_sets_private.cpp

```diff
@@ -1320,8 +1320,35 @@ TrustNode TheorySetsPrivate::ppRewrite(Node node,
   {
     case kind::SET_CHOOSE: return expandChooseOperator(node, lems);
     case kind::SET_IS_SINGLETON: return expandIsSingletonOperator(node);
-    default: return TrustNode::null();
+    case kind::SET_MINUS:
+    {
+      if (node[0].getKind() == kind::SET_UNIVERSE)
+      {
+        // Due to complications involving the cardinality graph, we must purify
+        // universe from argument of set minus, so that
+        //   (set.minus set.universe x)
+        // is replaced by
+        //   (set.minus univ x)
+        // along with the lemma (= univ set.universe), where univ is the
+        // purification skolem for set.universe. We require this purification
+        // since the cardinality graph incorrectly thinks that
+        // rewrite( (set.inter set.universe x) ), which evaluates to x, is
+        // a sibling of (set.minus set.universe x).
```

Comment:
Why is being siblings a problem here? Assuming only the presence of 3 terms `set.universe, x, (set.minus set.universe x)` the last 2 terms are indeed siblings and children of `set.universe`.

Reply:
We do not have special handling for set universe in the cardinality graph.  The only relationships are:   A\B is child of A and AvB, A^B is the child of A, A v B, B, and B\A is the child of B and AvB.

Since we use rewritten terms, (set.minus set.universe x) mistakenly thinks its sibling is x. This causes problems e.g. if x is equal to universe. The end result is that the universe set was having a child report that the universe set was a Venn region of itself.

Reply:
Discussed this offline.